src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -83,32 +83,101 @@ internal partial class CodeFixService : ForegroundThreadAffinitizedObject, ICode
 
             using (var diagnostics = SharedPools.Default<List<DiagnosticData>>().GetPooledObject())
             {
-                var fullResult = await _diagnosticService.TryAppendDiagnosticsForSpanAsync(document, range, diagnostics.Object, cancellationToken: cancellationToken).ConfigureAwait(false);
-                foreach (var diagnostic in diagnostics.Object)
-                {
-                    cancellationToken.ThrowIfCancellationRequested();
+                var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                var linkedToken = linkedTokenSource.Token;
 
-                    if (!range.IntersectsWith(diagnostic.TextSpan))
-                    {
-                        continue;
-                    }
+                var fullResult = await _diagnosticService.TryAppendDiagnosticsForSpanAsync(document, range, diagnostics.Object, cancellationToken: linkedToken).ConfigureAwait(false);
```

Comment:
1. what does 'fullResult' mean?  needs comment.
2. call this isFullResult as this is a boolean (that wasn't at all clear).